core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -326,6 +328,26 @@ public void run() {
     if (pruningEnabled && slashingProtectionContext.isPresent()) {
       scheduleAndExecuteInitialDbPruning();
     }
+    slashingProtectionContext.ifPresent(this::scheduleDbHealthCheck);
+  }
+
+  private void scheduleDbHealthCheck(final SlashingProtectionContext protectionContext) {
+    final DbHealthCheck dbHealthCheck =
+        new DbHealthCheck(
+            protectionContext, slashingProtectionParameters.getDbHealthCheckTimeoutMilliseconds());
+
+    Executors.newScheduledThreadPool(1)
```

Comment:
I think we might need to shut this thread pool down upon graceful stop of the web3signer process